extra/genentries

```diff
@@ -187,26 +227,62 @@ function readdesktopfile {
         fi
         if [ -z "$icon" ] && [ -d "$user_home/.local/share/icons" ]; then
-          icon=$(find -L "$user_home/.local/share/icons" -type f \( "${findargs[@]}" -false \) -print -quit)
+          icon=$(find -L "$user_home/.local/share/icons" -type f \( "${findargs[@]}" -false \) -print -quit 2>/dev/null)
         fi
         if [ -z "$icon" ] && [ -d "$user_home/.icons" ]; then
-          icon=$(find -L "$user_home/.icons" -type f \( "${findargs[@]}" -false \) -print -quit)
+          icon=$(find -L "$user_home/.icons" -type f \( "${findargs[@]}" -false \) -print -quit 2>/dev/null)
         fi
         if [ -z "$icon" ] && [ -d "/usr/share/icons" ]; then
-          icon=$(find -L "/usr/share/icons" -type f \( "${findargs[@]}" -false \) -print -quit)
+          icon=$(find -L "/usr/share/icons" -type f \( "${findargs[@]}" -false \) -print -quit 2>/dev/null)
+        fi
+        if [ -z "$icon" ] && [ -d "/usr/share/pixmaps/puppy" ]; then
+          icon=$(find -L "/usr/share/pixmaps/puppy" -type f \( "${findargs[@]}" -false \) -print -quit 2>/dev/null)
         fi
       fi
 
+      # if the icon is a SVG, xlunch cant show it, so convert to PNG and
+      # put it in /usr/share/xlunch/svgicons/<name>.png
       if [ "${icon: -4}" == ".svg" ]; then
-        if ! [ -f "$svgpath/"$iconname".png" ]; then
-          >&2 echo "Converting $icon to png"
+        # if the generated SVG doesnt exist yet
+        if [ ! -f "${svgpath}/${iconname}.png" ]; then
           mkdir -p "$svgpath"
-          convert -density 500 -resize "$iconsize"x"$iconsize" -background none "$icon" "$svgpath/"$iconname".png"
+
+          echo "$name:  $icon  =>  $svgpath/$iconname.png" >&2 
+
+          # use rsvg-convert to convert SVG to PNG
+          if [ ! -f "${svgpath}/${iconname}.png" ] && \
+             [ "$(which rsvg-convert)" != "" ];then
+            rsvg-convert "$icon" \
+              -w "$iconsize" \
+              -h "$iconsize" \
+              --keep-aspect-ratio \
+              --output "$svgpath/"$iconname".png" 2>/dev/null
+          fi
+
+          # use inkscape to convert SVG to PNG
+          if [ ! -f "${svgpath}/${iconname}.png" ] && \
+             [ "$(which inkscapelite)" != "" ];then
+            inkscapelite -z -e "${svgpath}/${iconname}.png" \
+            -w "$iconsize" \
+            -h "$iconsize" \
+            "$icon" 2>/dev/null
+          fi
+
+          # use convert (from ImageMagick)  if rsvg failed or not installed
+          if [ ! -f "${svgpath}/${iconname}.png" ] && \
+             [ "$(which convert)" != "" ];then
+            convert \
+              -density 500 \
+              -resize "$iconsize"x"$iconsize" \
+              -background none \
+              "$icon"
+          fi
         fi
-        icon="$svgpath/"$iconname".png"
+        # if a new PNG was created successfully, set it as our application icon
+        [ -f "${svgpath}/${iconname}.png" ] && icon="${svgpath}/${iconname}.png"
```

Comment:
Not sure if this is the behaviour we want. If the file was not created we should still use  the invalid path to let users know that it wasn't able to convert the icon.

Reply:
Your call, I have it there cos that is what works for me to not get any missing icons..

Reply:
Well, missing icons because of an error when converting shouldn't be an accepted outcome. It should create an error so the user is aware of the issue.